engine/telemetry/heartbeat.go

```diff
@@ -0,0 +1,87 @@
+package telemetry
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/dagger/dagger/engine/slog"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+)
+
+const HeartbeatInterval = 30 * time.Second
+
+// SpanHeartbeater is a SpanExporter that keeps track of live spans and
+// re-exports them periodically to the underlying SpanExporter to indicate that
+// they are indeed still live.
+type SpanHeartbeater struct {
```

Comment:
Previously heartbeating was a responsibility of each SDK's "live" OTel integration. Now it's just a layer we wrap around our Cloud exporter, and each SDK's minimal "live" implementation is just a SpanProcessor that calls OnEnd for every OnStart.